### PR TITLE
Support newline and tab literals in regex search-and-replace operations

### LINKED ISSUE #7645: Support for Replacing Text with Escaped Characters in Regex Mode
### Check for existing issues

- [X] Completed

### Describe the feature

In regex mode, while I can search for extended characters (such as newline "\n" and tab "\t") successfully, I cannot replace them with their escaped counterparts. For instance, if I want to remove all double empty lines in a document, I would typically search for "\n\n" and aim to replace it with a single newline "\n". However, zed currently inserts a literal "\n" string instead of interpreting it as a newline character.

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by mayfieldiv**
Closes #7645

Release Notes:

- Added support for inserting newlines (`\n`) and tabs (`\t`) in editor Regex search replacements ([#7645](https://github.com/zed-industries/zed/issues/7645)).

### CODE REVIEW

File: crates/project/src/search.rs [hunk 1]
```diff
@@ -231,6 +231,15 @@ impl SearchQuery {
                 regex, replacement, ..
             } => {
                 if let Some(replacement) = replacement {
+                    let replacement = Regex::new(r"\\\\|\\n|\\t").unwrap().replace_all(
```
Comment (osiewicz) [inline crates/project/src/search.rs:234]: Let's store that Regex object in something like OnceCell/[OnceLock](https://doc.rust-lang.org/std/sync/struct.OnceLock.html).

File: crates/project/src/search.rs [hunk 2]
```diff
@@ -231,6 +231,15 @@ impl SearchQuery {
...
+                        replacement,
+                        |c: &Captures| match c.get(0).unwrap().as_str() {
+                            r"\\" => "\\",
+                            r"\n" => "\n",
+                            r"\t" => "\t",
+                            x => unreachable!("Unexpected escape sequence: {}", x),
```
Comment (osiewicz) [inline crates/project/src/search.rs:240]:
```suggestion
                        |c: &Captures| match c.get(0).map(Match::as_str) {
                            Some(r"\\") => "\\",
                            Some(r"\n") => "\n",
                            Some(r"\t") => "\t",
                            x => unreachable!("Unexpected escape sequence: {:?}", x),
```

File: crates/project/src/search.rs (hunk 1 above)
Comment (mayfieldiv) [reply crates/project/src/search.rs]: sure thing. done

**Comment by osiewicz** [comment]
Thanks!

**Pull request merged**